### ISSUE TITLE
about the decoder of prott5 model

### ISSUE THREAD

**Issue by xiebaoshu058** [agemagician/ProtTrans#92]
Hello, I'm a greenhand of machine learning and my question may be a little stupid. I wonder why the outputs of the prott5 decoder are still features? From other websites I learned that the t5 model has a linear and softmax module behind the decoder, that's the question: in your project, does prott5 has a linear and softmax module while training? If yes, how can I get the pretrained linear module? If no, can I train my own linear module on my own dataset with the decoder module fixed?
Thank you very much for answering my question!

**Comment by mheinzinger** [comment]
Yes, the decoder still has "features" or embeddings in a a sense that it also produces hidden states which can be interpreted as features. However, our experiments showed that embeddings/features derived from the encoder-side of ProtT5 perform much worse, which is why we usually use only embeddings from ProtT5's encoder.
If you are interested in the linear layer/softmax, you should load ProtT5 using this module: https://huggingface.co/docs/transformers/v4.23.1/en/model_doc/t5#transformers.T5ForConditionalGeneration

**Comment by xiebaoshu058** [comment]
@mheinzinger I think you mean  that the features derived from the decoder perform much worse?
thank you so much for replying so quickly, I'm a big fan of this project!

**Comment by mheinzinger** [comment]
Absolutely, sorry. Fixed it in my comment above
